prometheus_client/openmetrics/exposition.py

```diff
@@ -58,44 +81,80 @@ def generate_latest(registry, escaping=UNDERSCORES):
                             for k, v in items])
                 if labelstr:
                     labelstr = "{" + labelstr + "}"
-
                 if s.exemplar:
-                    if not _is_valid_exemplar_metric(metric, s):
-                        raise ValueError(f"Metric {metric.name} has exemplars, but is not a histogram bucket or counter")
-                    labels = '{{{0}}}'.format(','.join(
-                        ['{}="{}"'.format(
-                            k, v.replace('\\', r'\\').replace('\n', r'\n').replace('"', r'\"'))
-                            for k, v in sorted(s.exemplar.labels.items())]))
-                    if s.exemplar.timestamp is not None:
-                        exemplarstr = ' # {} {} {}'.format(
-                            labels,
-                            floatToGoString(s.exemplar.value),
-                            s.exemplar.timestamp,
-                        )
-                    else:
-                        exemplarstr = ' # {} {}'.format(
-                            labels,
-                            floatToGoString(s.exemplar.value),
-                        )
+                    exemplarstr = _compose_exemplar_string(metric, s, s.exemplar)
                 else:
                     exemplarstr = ''
                 timestamp = ''
                 if s.timestamp is not None:
                     timestamp = f' {s.timestamp}'
+                
+                native_histogram = ''
+                negative_spans = ''
+                negative_deltas = ''
+                positive_spans = ''
+                positive_deltas = ''
+                     
+                if s.native_histogram:
+                    # Initialize basic nh template
+                    nh_sample_template = '{{count:{},sum:{},schema:{},zero_threshold:{},zero_count:{}'
+
+                    args = [
+                        s.native_histogram.count_value,
+                        s.native_histogram.sum_value,
+                        s.native_histogram.schema,
+                        s.native_histogram.zero_threshold,
+                        s.native_histogram.zero_count,
+                    ]
+                  
+                    # If there are neg spans, append them and the neg deltas to the template and args
+                    if s.native_histogram.neg_spans:
+                        negative_spans = ','.join([f'{ns[0]}:{ns[1]}' for ns in s.native_histogram.neg_spans])
+                        negative_deltas = ','.join(str(nd) for nd in s.native_histogram.neg_deltas)
+                        nh_sample_template += ',n_spans:[{}]'
```

Comment:
Spans and buckets are prefixed with `positive` and `negative` in the proposal, not `n`, `p`. Same in OM2.0 proposal. https://github.com/prometheus/proposals/blob/main/proposals/0032-native_histograms_text_format.md#how